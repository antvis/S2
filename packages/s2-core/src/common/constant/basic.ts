import { i18n } from '../i18n';
// 值字段的 id 是固定的！
export const VALUE_FIELD = '$$value$$';
export const EXTRA_FIELD = '$$extra$$';
export const TOTAL_VALUE = '$$total$$';
export const SERIES_NUMBER_FIELD = '$$series_number$$';
export const DEFAULT_CORNER_TEXT = i18n('指标');

export const BACK_GROUND_GROUP_CONTAINER_Z_INDEX = 0;

// foregroundGroup 上的 children 层叠顺序
export const FRONT_GROUND_GROUP_CONTAINER_Z_INDEX = 3; // 约定这个 z-index 为 0 的 container 作为基准
export const FRONT_GROUND_GROUP_COL_SCROLL_Z_INDEX = 3;
export const FRONT_GROUND_GROUP_COL_FROZEN_Z_INDEX = 4;
export const FRONT_GROUND_GROUP_RESIZE_AREA_Z_INDEX = 5;
export const FRONT_GROUND_GROUP_BRUSH_SELECTION_Z_INDEX = 5;

// panelGroup 上的 children 层叠顺序
export const PANEL_GROUP_GROUP_CONTAINER_Z_INDEX = 1;
export const PANEL_GROUP_SCROLL_GROUP_Z_INDEX = 1;
export const PANEL_GROUP_FROZEN_GROUP_Z_INDEX = 2;
export const PANEL_GROUP_HOVER_BOX_GROUP_Z_INDEX = 1;

// group's key
export const KEY_GROUP_BACK_GROUND = 'backGroundGroup';
export const KEY_GROUP_FORE_GROUND = 'foreGroundGroup';
export const KEY_GROUP_PANEL_GROUND = 'panelGroup';
export const KEY_GROUP_PANEL_SCROLL = 'panelScrollGroup';
export const KEY_GROUP_PANEL_FROZEN_ROW = 'frozenRowGroup';
export const KEY_GROUP_PANEL_FROZEN_COL = 'frozenColGroup';
export const KEY_GROUP_PANEL_FROZEN_TRAILING_ROW = 'frozenTrailingRowGroup';
export const KEY_GROUP_PANEL_FROZEN_TRAILING_COL = 'frozenTrailingColGroup';
export const KEY_GROUP_PANEL_FROZEN_TOP = 'frozenTopGroup';
export const KEY_GROUP_PANEL_FROZEN_BOTTOM = 'frozenBottomGroup';
export const KEY_GROUP_ROW_RESIZE_AREA = 'rowResizeAreaGroup';
export const KEY_GROUP_FROZEN_ROW_RESIZE_AREA = 'rowFrozenResizeAreaGroup';
export const KEY_GROUP_FROZEN_SPLIT_LINE = 'frozenSplitLine';
export const KEY_GROUP_ROW_INDEX_RESIZE_AREA = 'rowIndexResizeAreaGroup';
export const KEY_GROUP_CORNER_RESIZE_AREA = 'cornerResizeAreaGroup';
export const KEY_GROUP_COL_RESIZE_AREA = 'colResizeAreaGroup';
export const KEY_GROUP_FROZEN_COL_RESIZE_AREA = 'colFrozenResizeAreaGroup';
export const KEY_GROUP_COL_HORIZONTAL_RESIZE_AREA =
  'colHorizontalResizeAreaGroup';
export const KEY_GROUP_COL_SCROLL = 'colScrollGroup';
export const KEY_GROUP_COL_FROZEN = 'colFrozenGroup';
export const KEY_GROUP_COL_FROZEN_TRAILING = 'colFrozenTrailingGroup';

// key of series number node in corner header
export const KEY_SERIES_NUMBER_NODE = 'series-number-node';

export const HORIZONTAL_RESIZE_AREA_KEY_PRE = 'horizontal-resize-area-';
export const TABLE_COL_HORIZONTAL_RESIZE_AREA_KEY =
  'table-col-horizontal-resize-area';

export const KEY_COL_REAL_WIDTH_INFO = 'col-real-width-info';

export const ROOT_ID = 'root';
export const ID_SEPARATOR = '[&]';
export const EMPTY_PLACEHOLDER = '-';

export const PADDING_TOP = 0;
export const PADDING_RIGHT = 1;
export const PADDING_DOWN = 2;
export const PADDING_LEFT = 3;

export const MIN_CELL_WIDTH = 28;
export const MIN_CELL_HEIGHT = 16;

// data precision
export const PRECISION = 16;

export const ROOT_BEGINNING_REGEX = /^root\[&\]*/;

export const IMAGE = 'image';

// 角头最大占整个容器的比例 (0-1)
export const CORNER_MAX_WIDTH_RATIO = 0.5;
