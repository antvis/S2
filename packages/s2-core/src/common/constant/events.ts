export { InteractionEvent } from './interaction';

// event keys
export const KEY_LIST_SORT = 'spreadsheet:list-sort';
export const KEY_COLLAPSE_TREE_ROWS = 'spreadsheet:collapsed-tree-rows';
export const KEY_JUMP_HREF = 'spreadsheet:jump-href';
export const KEY_AFTER_HEADER_LAYOUT = 'spreadsheet:after-header-layout';
export const KEY_COL_NODE_BORDER_REACHED =
  'spreadsheet:col-node-border-reached';
export const KEY_ROW_NODE_BORDER_REACHED =
  'spreadsheet:row-node-border-reached';
export const KEY_TREE_ROWS_COLLAPSE_ALL = 'spreadsheet:toggle-collapse-all';
export const KEY_CELL_SCROLL = 'spreadsheet:cell-scroll';
export const KEY_PAGINATION = 'spreadsheet:pagination';

// x-report to ss
export const KEY_UPDATE_PROPS = 'spreadsheet-update-props';
export const KEY_COLLAPSE_ROWS = 'spreadsheet:collapsed-rows';
export const KEY_AFTER_COLLAPSE_ROWS = 'spreadsheet:after-collapsed-rows';

export enum ResizeEventType {
  ROW_W = 'spreadsheet:change-row-header-width',
  COL_W = 'spreadsheet:change-column-header-width',
  ROW_H = 'spreadsheet:change-row-header-height',
  COL_H = 'spreadsheet:change-column-header-height',
  TREE_W = 'spreadsheet:change-tree-width',
}

export enum S2Event {
  COL_CELL_CLICK = 'col-cell:click',
  ROW_CELL_CLICK = 'row-cell:click',
  DATA_CELL_CLICK = 'data-cell:click',
  CORNER_CELL_CLICK = 'corner-cell:click',
  MERGED_CELLS_CLICK = 'merged-cells:click',

  COL_CELL_HOVER = 'col-cell:hover',
  ROW_CELL_HOVER = 'row-cell:hover',
  DATA_CELL_HOVER = 'data-cell:hover',
  CORNER_CELL_HOVER = 'corner-cell:hover',
  MERGED_CELLS_HOVER = 'merged-cells:hover',

  COL_CELL_MOUSE_DOWN = 'col-cell:mouse-down',
  ROW_CELL_MOUSE_DOWN = 'row-cell:mouse-down',
  DATA_CELL_MOUSE_DOWN = 'data-cell:mouse-down',
  CORNER_CELL_MOUSE_DOWN = 'corner-cell:mouse-down',
  MERGED_CELLS_MOUSE_DOWN = 'merged-cells:mouse-down',

  COL_CELL_MOUSE_UP = 'col-cell:mouse-up',
  ROW_CELL_MOUSE_UP = 'row-cell:mouse-up',
  DATA_CELL_MOUSE_UP = 'data-cell:mouse-up',
  CORNER_CELL_MOUSE_UP = 'corner-cell:mouse-up',
  MERGED_CELLS_MOUSE_UP = 'merged-cells:mouse-up',

  COL_CELL_MOUSE_MOVE = 'col-cell:mouse-move',
  ROW_CELL_MOUSE_MOVE = 'row-cell:mouse-move',
  DATA_CELL_MOUSE_MOVE = 'data-cell:mouse-move',
  CORNER_CELL_MOUSE_MOVE = 'corner-cell:mouse-move',
  MERGED_ELLS_MOUSE_MOVE = 'merged-cells:mouse-move',

  GLOBAL_RESIZE_MOUSE_DOWN = 'global:resize:mouse-down',
  GLOBAL_RESIZE_MOUSE_MOVE = 'global:resize:mouse-move',
  GLOBAL_RESIZE_MOUSE_UP = 'global:resize-mouse-up',

  GLOBAL_KEYBOARD_DOWN = 'global:keyboard-down',
  GLOBAL_KEYBOARD_UP = 'global:keyboard-up',
  GLOBAL_COPIED = 'global:copied',
  GLOBAL_MOUSE_UP = 'global:mouse-up',

  GLOBAL_CLEAR_INTERACTION_STYLE_EFFECT = 'global:clear-interaction-style-effect',
}

export enum OriginEventType {
  MOUSE_DOWN = 'mousedown',
  MOUSE_MOVE = 'mousemove',
  MOUSE_UP = 'mouseup',
  KEY_DOWN = 'keydown',
  KEY_UP = 'keyup',
  CLICK = 'click',
}

export enum InterceptEventType {
  HOVER = 'hover',
  CLICK = 'click',
}

export type InterceptEvent =
  | InterceptEventType.HOVER
  | InterceptEventType.CLICK;
