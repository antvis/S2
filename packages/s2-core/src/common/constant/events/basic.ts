export enum S2Event {
  /** ================ Row Cell ================  */
  ROW_CELL_HOVER = 'row-cell:hover',
  ROW_CELL_CLICK = 'row-cell:click',
  ROW_CELL_DOUBLE_CLICK = 'row-cell:double-click',
  ROW_CELL_MOUSE_DOWN = 'row-cell:mouse-down',
  ROW_CELL_MOUSE_UP = 'row-cell:mouse-up',
  ROW_CELL_MOUSE_MOVE = 'row-cell:mouse-move',
  ROW_CELL_COLLAPSE_TREE_ROWS = 'row-cell:collapsed-tree-rows',

  /** ================ Col Cell ================  */
  COL_CELL_HOVER = 'col-cell:hover',
  COL_CELL_CLICK = 'col-cell:click',
  COL_CELL_DOUBLE_CLICK = 'col-cell:double-click',
  COL_CELL_MOUSE_DOWN = 'col-cell:mouse-down',
  COL_CELL_MOUSE_UP = 'col-cell:mouse-up',
  COL_CELL_MOUSE_MOVE = 'col-cell:mouse-move',

  /** ================ Data Cell ================  */
  DATA_CELL_HOVER = 'data-cell:hover',
  DATA_CELL_CLICK = 'data-cell:click',
  DATA_CELL_DOUBLE_CLICK = 'data-cell:double-click',
  DATA_CELL_MOUSE_UP = 'data-cell:mouse-up',
  DATA_CELL_MOUSE_DOWN = 'data-cell:mouse-down',
  DATA_CELL_MOUSE_MOVE = 'data-cell:mouse-move',
  DATA_CELL_TREND_ICON_CLICK = 'data-cell:trend-icon-click',
  DATE_CELL_BRUSH_SELECTION = 'data-cell:brush-selection',

  /** ================ Corner Cell ================  */
  CORNER_CELL_HOVER = 'corner-cell:hover',
  CORNER_CELL_CLICK = 'corner-cell:click',
  CORNER_CELL_DOUBLE_CLICK = 'corner-cell:double-click',
  CORNER_CELL_MOUSE_DOWN = 'corner-cell:mouse-down',
  CORNER_CELL_MOUSE_UP = 'corner-cell:mouse-up',
  CORNER_CELL_MOUSE_MOVE = 'corner-cell:mouse-move',

  /** ================ Merged Cells ================  */
  MERGED_CELLS_HOVER = 'merged-cells:hover',
  MERGED_CELLS_CLICK = 'merged-cells:click',
  MERGED_CELLS_DOUBLE_CLICK = 'merged-cells:double-click',
  MERGED_CELLS_MOUSE_DOWN = 'merged-cells:mouse-down',
  MERGED_CELLS_MOUSE_UP = 'merged-cells:mouse-up',
  MERGED_CELLS_MOUSE_MOVE = 'merged-cells:mouse-move',

  /** ================ Sort ================  */
  RANGE_SORT = 'sort:range-sort',
  RANGE_SORTED = 'sort:range-sorted',

  /** ================ Table Filter ================  */
  RANGE_FILTER = 'filter:range-filter',
  RANGE_FILTERED = 'filter:range-filtered',

  /** ================ Table Layout ================  */
  LAYOUT_AFTER_HEADER_LAYOUT = 'layout:after-header-layout',
  LAYOUT_CELL_SCROLL = 'layout:cell-scroll',
  LAYOUT_PAGINATION = 'layout:pagination',
  LAYOUT_COLLAPSE_ROWS = 'layout:collapsed-rows',
  LAYOUT_AFTER_COLLAPSE_ROWS = 'layout:after-collapsed-rows',
  LAYOUT_TREE_ROWS_COLLAPSE_ALL = 'layout:toggle-collapse-all',
  LAYOUT_COLS_EXPANDED = 'layout:table-col-expanded',
  LAYOUT_COLS_HIDDEN = 'layout:table-col-hidden',
  LAYOUT_AFTER_RENDER = 'layout:after-render',
  LAYOUT_BEFORE_RENDER = 'layout:before-render',
  LAYOUT_DESTROY = 'layout:destroy',

  /** ================ Table Layout Resize ================  */
  LAYOUT_RESIZE = 'layout:resize',
  LAYOUT_RESIZE_SERIES_WIDTH = 'layout:resize-series-width',
  LAYOUT_RESIZE_ROW_WIDTH = 'layout:resize-row-width',
  LAYOUT_RESIZE_ROW_HEIGHT = 'layout:resize-row-height',
  LAYOUT_RESIZE_COL_WIDTH = 'layout:resize-column-width',
  LAYOUT_RESIZE_COL_HEIGHT = 'layout:resize-column-height',
  LAYOUT_RESIZE_TREE_WIDTH = 'layout:resize-tree-width',
  LAYOUT_RESIZE_MOUSE_DOWN = 'layout:resize:mouse-down',
  LAYOUT_RESIZE_MOUSE_MOVE = 'layout:resize:mouse-move',
  LAYOUT_RESIZE_MOUSE_UP = 'layout:resize-mouse-up',

  /** ================ Global ================  */
  GLOBAL_KEYBOARD_DOWN = 'global:keyboard-down',
  GLOBAL_KEYBOARD_UP = 'global:keyboard-up',
  GLOBAL_COPIED = 'global:copied',
  GLOBAL_MOUSE_UP = 'global:mouse-up',
  GLOBAL_MOUSE_MOVE = 'global:mouse-move',
  GLOBAL_ACTION_ICON_CLICK = 'global:action-icon-click',
  GLOBAL_ACTION_ICON_HOVER = 'global:action-icon-hover',
  GLOBAL_CONTEXT_MENU = 'global:context-menu',
  GLOBAL_SELECTED = 'global:selected',
  GLOBAL_HOVER = 'global:hover',
  GLOBAL_RESET = 'global:reset',
  GLOBAL_LINK_FIELD_JUMP = 'global:link-field-jump',
}
