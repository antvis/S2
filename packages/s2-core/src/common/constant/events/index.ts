export * from './interaction';
export * from './origin';
export * from './basic';
