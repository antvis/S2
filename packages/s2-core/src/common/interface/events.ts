// 这里存放 emit 事件 透出的信息

import { Event as GEvent } from '@antv/g-canvas';
import { S2CellType } from './interaction';
import { Node } from '@/facet/layout/node';

export interface CellScrollPosition {
  scrollX: number;
  scrollY: number;
  thumbOffset?: number;
}
export interface ListSortParams {
  sortFieldId: string;
  sortMethod: string;
}

export type LayoutRow = [number, string, string];

export type LayoutCol = [number, string, string];

export interface TargetCellInfo {
  target: S2CellType;
  event: GEvent;
  viewMeta: Node;
}
