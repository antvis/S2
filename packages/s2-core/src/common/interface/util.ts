export type DeepRequired<T extends Record<string, any>> = {
  [K in keyof T]-?: NonNullable<T[K]> extends Record<string, any>
    ? DeepRequired<NonNullable<T[K]>>
    : NonNullable<T[K]>;
};
