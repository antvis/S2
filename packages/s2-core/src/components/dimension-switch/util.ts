import { DimensionType, DimensionItem } from './dimension';

export interface OperatedType {
  [type: string]: string[];
}

export const getDimensionsByPredicate = (
  data: DimensionType[],
  predicate: (value: DimensionItem) => boolean,
) => {
  return data.reduce((obj, t) => {
    obj[t.type] = t.items.filter(predicate).map((i) => i.id);
    return obj;
  }, {} as OperatedType);
};
