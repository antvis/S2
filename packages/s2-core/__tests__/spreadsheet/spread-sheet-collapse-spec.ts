import * as mockDataConfig from 'tests/data/simple-data.json';
import { getContainer } from 'tests/util/helpers';
import type { RowCellCollapsedParams } from '../../src/common/interface';
import { S2Event } from './../../src/common/constant/events/basic';
import { PivotSheet, SpreadSheet } from '@/sheet-type';
import type { Node } from '@/facet/layout/node';

describe('SpreadSheet Collapse/Expand Tests', () => {
  let container: HTMLElement;
  let s2: SpreadSheet;

  const mapNodes = (spreadsheet: SpreadSheet) =>
    spreadsheet.facet.getRowLeafNodes().map((node) => node.id);

  beforeEach(async () => {
    container = getContainer();
    s2 = new PivotSheet(
      container,
      {
        ...mockDataConfig,
        fields: {
          rows: ['province', 'city', 'type'],
          columns: [],
          values: ['price', 'cost'],
          valueInCols: true,
        },
      },
      {
        width: 600,
        height: 200,
        hierarchyType: 'tree',
        style: {
          rowCell: {
            expandDepth: undefined,
          },
        },
      },
    );
    await s2.render();
  });

  afterEach(() => {
    s2.destroy();
  });

  describe('Tree Mode', () => {
    test('should init rows with expandDepth config', async () => {
      s2.setOptions({
        style: {
          rowCell: {
            expandDepth: 0,
          },
        },
      });
      await s2.render();

      const rowLeafNodes = s2.facet.getRowLeafNodes();

      expect(rowLeafNodes).toHaveLength(3);
      expect(mapNodes(s2)).toMatchInlineSnapshot(`
              Array [
                "root[&]浙江",
                "root[&]浙江[&]义乌",
                "root[&]浙江[&]杭州",
              ]
          `);

      s2.setOptions({
        style: {
          rowCell: {
            expandDepth: 1,
          },
        },
      });
      await s2.render();

      expect(mapNodes(s2)).toMatchInlineSnapshot(`
              Array [
                "root[&]浙江",
                "root[&]浙江[&]义乌",
                "root[&]浙江[&]义乌[&]笔",
                "root[&]浙江[&]杭州",
                "root[&]浙江[&]杭州[&]笔",
              ]
          `);
    });

    test('should collapse all row nodes', async () => {
      s2.setOptions({
        style: {
          rowCell: {
            collapseAll: true,
          },
        },
      });

      await s2.render(false);

      expect(mapNodes(s2)).toMatchInlineSnapshot(`
              Array [
                "root[&]浙江",
              ]
          `);
    });

    test('should collapse by field', async () => {
      s2.setOptions({
        style: {
          rowCell: {
            collapseFields: {
              province: true,
            },
          },
        },
      });

      await s2.render(false);

      expect(mapNodes(s2)).toMatchInlineSnapshot(`
              Array [
                "root[&]浙江",
              ]
          `);
    });

    test('should collapse by field id', async () => {
      s2.setOptions({
        style: {
          rowCell: {
            collapseFields: {
              'root[&]浙江[&]义乌': true,
            },
          },
        },
      });

      await s2.render();

      expect(mapNodes(s2)).toMatchInlineSnapshot(`
              Array [
                "root[&]浙江",
                "root[&]浙江[&]义乌",
                "root[&]浙江[&]杭州",
                "root[&]浙江[&]杭州[&]笔",
              ]
          `);
    });

    test('should collapse use collapseFields first', async () => {
      s2.setOptions({
        style: {
          rowCell: {
            collapseAll: false,
            collapseFields: {
              city: true,
            },
          },
        },
      });

      await s2.render(false);

      expect(mapNodes(s2)).toMatchInlineSnapshot(`
              Array [
                "root[&]浙江",
                "root[&]浙江[&]义乌",
                "root[&]浙江[&]杭州",
              ]
          `);
    });

    test('should collapse use expandDepth first', async () => {
      s2.setOptions({
        style: {
          rowCell: {
            collapseAll: true,
            expandDepth: 1,
          },
        },
      });
      await s2.render(false);

      expect(mapNodes(s2)).toMatchInlineSnapshot(`
              Array [
                "root[&]浙江",
                "root[&]浙江[&]义乌",
                "root[&]浙江[&]义乌[&]笔",
                "root[&]浙江[&]杭州",
                "root[&]浙江[&]杭州[&]笔",
              ]
          `);
    });

    test('should collapse use collapseFields first when contain collapseAll and expandDepth config', async () => {
      s2.setOptions({
        style: {
          rowCell: {
            collapseAll: false,
            collapseFields: {
              'root[&]浙江[&]杭州': true,
            },
            expandDepth: null,
          },
        },
      });
      await s2.render(false);

      expect(mapNodes(s2)).toMatchInlineSnapshot(`
              Array [
                "root[&]浙江",
                "root[&]浙江[&]义乌",
                "root[&]浙江[&]义乌[&]笔",
                "root[&]浙江[&]杭州",
              ]
          `);
    });

    test('should collapse use collapseFields by node id first', async () => {
      s2.setOptions({
        style: {
          rowCell: {
            collapseFields: {
              province: false,
              'root[&]浙江': true,
            },
          },
        },
      });
      await s2.render(false);

      expect(mapNodes(s2)).toMatchInlineSnapshot(`
        Array [
          "root[&]浙江",
        ]
      `);
    });

    test('should collapse all nodes if collapseAll is true and collapseFields is undefined', async () => {
      s2.setOptions({
        style: {
          rowCell: {
            collapseAll: true,
            collapseFields: undefined,
          },
        },
      });
      await s2.render(false);

      expect(mapNodes(s2)).toMatchInlineSnapshot(`
        Array [
          "root[&]浙江",
        ]
      `);
    });

    test('should emit collapse event', () => {
      const onCollapsed = jest.fn();

      s2.on(S2Event.ROW_CELL_COLLAPSED, onCollapsed);

      const node = { id: 'testId' } as unknown as Node;
      const treeRowType: RowCellCollapsedParams = {
        isCollapsed: false,
        node,
      };

      const params: RowCellCollapsedParams = {
        isCollapsed: false,
        collapseFields: {
          [node.id]: false,
        },
        node,
      };

      s2.emit(S2Event.ROW_CELL_COLLAPSED__PRIVATE, treeRowType);

      expect(onCollapsed).toHaveBeenCalledWith(params);
    });

    test('should emit collapse all event', () => {
      const onCollapsed = jest.fn();

      s2.on(S2Event.ROW_CELL_ALL_COLLAPSED, onCollapsed);

      s2.emit(S2Event.ROW_CELL_ALL_COLLAPSED__PRIVATE, false);

      expect(onCollapsed).toHaveBeenCalledWith(true);
    });
  });
});
