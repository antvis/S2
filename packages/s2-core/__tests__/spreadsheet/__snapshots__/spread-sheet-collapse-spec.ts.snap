// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpreadSheet Collapse/Expand Tests Tree Mode should only expand current group row nodes 1`] = `
Array [
  "root[&]浙江省",
  "root[&]浙江省[&]杭州市",
  "root[&]浙江省[&]绍兴市",
  "root[&]浙江省[&]宁波市",
  "root[&]浙江省[&]舟山市",
  "root[&]四川省",
]
`;

exports[`SpreadSheet Collapse/Expand Tests Tree Mode should only expand current group row nodes for custom tree 1`] = `
Array [
  "root[&]a-1",
  "root[&]a-1[&]a-1-1",
  "root[&]a-1[&]a-1-2",
  "root[&]a-2",
  "root[&]a-3",
]
`;

exports[`SpreadSheet Collapse/Expand Tests Tree Mode should sync corner cell collapse all icon status 1`] = `
Array [
  "root[&]浙江省",
  "root[&]四川省",
]
`;
